src/kafka_fdw.c

```diff
@@ -1095,6 +1109,16 @@ kafkaBeginForeignModify(ModifyTableState *mtstate,
         }
     }
 
+    /* If internal column is at the end, this may still require deletion after.
+     * So we will do that here.
+     */
+    if (del)
+    {
+         festate->attnumlist = list_delete_cell(festate->attnumlist, del, delprev);
+         del  = NULL;
+         delprev = NULL;
+    }
+
     conf = rd_kafka_conf_new();
```

Comment:
for me

```
for (lc = list_head(festate->attnumlist); lc; lc = next)
    {
        int attnum = lfirst_int(lc);
        next = lnext(lc);

        if (!parsable_attnum(attnum, kafka_options))
        {
            festate->attnumlist = list_delete_cell(festate->attnumlist, lc, prev);
        }
        else
        {

            Form_pg_attribute attr = RelationGetDescr(rel)->attrs[attnum - 1];
            Assert(!attr->attisdropped);

            getTypeOutputInfo(attr->atttypid, &typefnoid, &isvarlena);
            fmgr_info(typefnoid, &festate->out_functions[num]);

            if (parse_options.format == JSON)
            {
                festate->attnames[num]    = getJsonAttname(attr, &festate->attname_buf);
                festate->typioparams[num] = attr->atttypid;
                DEBUGLOG("type oid %u", attr->atttypid);
            }

            num++;
            prev = lc;
        }
    }
```

feels more understandable which would also better align with the foreach macro in `pg_list.h`
```
#define foreach(cell, l)	\
	for ((cell) = list_head(l); (cell) != NULL; (cell) = lnext(cell))
```